specification/sbi_atee.adoc

```diff
@@ -13,6 +44,953 @@ running a VMM).
+. sbi_tee_tsm_local_fence()
+
+The memory conversion process is complete when sbi_tee_tsm_local_fence() is successfully completed
+on the every VCPU on the platform.
+
+Converted memory may be assigned to TVMs, but cannot be repurposed for regular use unless it's reclaimed.
+
+=== TVM creation
+
+TVMs are created using the sbi_tee_create_tvm(). This creates a TVM, and sets with state "initializing". The
+host must assign confidential memory for page tables, payload mapping, and VCPUs before it can be transitioned
+into a "runnable" state.
+
+=== TVM memory management
+
+The host is responsible for the following:
+
+. Dontations for the page-table pool
+.. The host must ensure that the TSM has sufficient confidential memory for mapping and
+   managing TVM page-tables for the code and data payloads (sbi_tee_tvm_add_page_table_pages())
+. Defining memory regions for confidential, MMIO, and shared memory regions
+.. The host must declare the TVM physical address ranges for mapping of confidential and
+   non-confidential memory. There can be multiples ranges of each type, but no two regions
+   can overlap. Both confidential and non-confidential regions can be sparsely mapped; however,
+   any sparsely mapped confidential page that's demand-page can only be a demand-zero page. 
+   The `sbi_tee_tvm_add_confidential_memory_region()`, `sbi_tee_tvm_add_emulated_mmio_region()`,
+   and `sbi_tee_tvm_add_shared_memory_region()` can be used for this purpose.
+. Mapping of code and data payload to the TVM
+.. The host must copy the TVM code and data payload from non-confidential memory to confidential
+   memory by calling `sbi_tee_tvm_add_measured_pages()`
+. Mapping of zero-page confidential pages to the TVM
+.. The host can map zero-pages in a TVM confidential memory range at any point of time by
+   calling `sbi_tee_tvm_add_zero_pages()`.
+. Mapping non-confidential shared pages on demand
+.. The host can share non-confidential memory with the TVM by calling `sbi_tee_tvm_add_shared_pages()`
+
+
+=== VCPU assignment
+
+The host must register platform VCPUs with the TSM before they can be used for TVM executio by calling
+`sbi_tee_tvm_vcpu_create()`.
+
+=== TVM execution
+
+Following assignment of memory and VCPU resources, the host can transition the guest into a "runnable"
+state by calling `sbi_tee_tvm_finalize()`. Note that some TEE calls are no longer permissble after this
+transition.
+
+The host can use the `sbi_tee_tvm_vcpu_set_register()` to setup TVM execution parameters like the entry
+point, and then call `sbi_tee_tvm_vcpu_run()` to begin execution. TVM execution continues until there an
+event like an interrupt, or fault that cannot be serviced by the TSM. The host can examine the return code
+from the ECALL, and use `sbi_tee_tvm_get_vcpu_register()` to determine further course of action. This may
+involve serving page-faults for demand-page regions, servicing MMIO requests, or terminating the TVM in
+case of an irrecoverable exception.
```

Comment:
We should get rid of entryPC & entryArg once we have the shared memory based approach in place to replace set/get registers.

Reply:
Yes -- see Andrew's PR here: https://github.com/rivosinc/salus/pull/96

Reply:
There was also some discussion to make finalize an implicit op as part of the first "vcpu run"